server/index.ts

```diff
@@ -1,29 +1,77 @@
-import * as http from 'http';
-import socketIo from 'socket.io';
-import * as socketFunctions from './socketFunctions';
-import { mongoose } from '@typegoose/typegoose';
-import { exit } from 'process';
-import { seedCards } from './seedCards';
-import { Shape, FillStyle, Color } from './db/card';
+import * as http from "http";
+import socketIo from "socket.io";
+import { mongoose } from "@typegoose/typegoose";
+import { exit } from "process";
+import { seedCards } from "./seedCards";
+import { Shape, FillStyle, Color } from "./db/card";
+import { RoomController } from "./controllers/roomController";
 
 const PORT = process.env.PORT || 3000;
-const MONGODB_URI = process.env.MONGODB_URI || 'mongodb://localhost:27017/passt';
+const MONGODB_URI =
+  process.env.MONGODB_URI || "mongodb://localhost:27017/passt";
 if (!MONGODB_URI) {
-    console.error('Please set the MONGODB_URI environment variable');
-    exit(1);
+  console.error("Please set the MONGODB_URI environment variable");
+  exit(1);
 }
 
 const app = http.createServer();
 const io = socketIo(app);
+const roomController = new RoomController();
 
-io.on('connection', (socket) => {
-    socket.on('join', (data: socketFunctions.JoinRoomData) => socketFunctions.joinRoom)
-})
+io.on("connection", (socket) => {
+  socket.on("create", (data: { open: boolean }) =>
+    roomController
+      .createRoom(data.open)
+      .then((room) => socket.emit("createdRoom", room))
+  );
+  socket.on("join", (data: { roomCode: string; playerName: string }) =>
+    roomController
+      .joinRoom(data.roomCode, socket.id, data.playerName)
+      .then(({ room, player }) => {
+        socket.join(room.roomCode);
+        socket.broadcast.to(room.roomCode).emit("new_player", player);
+        socket.emit("joined_successfully", room);
```

Comment:
Also need to send new player the current state of the room to initialize their client UI

Reply:
This is in case they join an ongoing game

Reply:
Should we allow for them to be able to join an ongoing game?

Reply:
I think so since a game can be picked up at any time. If not, we would need to present the user a waiting screen until the next game starts.

Reply:
If they join late, though, they'll be behind and are unlikely to win. I would vote in favor of a waiting screen, personally, as that's consistent with other games.